src/core/presentation_definition.rs

```diff
@@ -0,0 +1,382 @@
+use super::credential_format::*;
+use super::input_descriptor::*;
+use super::presentation_submission::*;
+
+use std::collections::HashMap;
+
+use anyhow::{bail, Result};
+use serde::{Deserialize, Serialize};
+use serde_json::Map;
+use ssi_claims::jwt::VerifiablePresentation;
+
+/// A presentation definition is a JSON object that describes the information a [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) requires of a [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder).
+///
+/// > Presentation Definitions are objects that articulate what proofs a [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) requires.
+/// > These help the [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) to decide how or whether to interact with a [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder).
+///
+/// Presentation Definitions are composed of inputs, which describe the forms and details of the
+/// proofs they require, and optional sets of selection rules, to allow [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder)s flexibility
+/// in cases where different types of proofs may satisfy an input requirement.
+///
+/// For more information, see: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition)
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+pub struct PresentationDefinition {
+    id: String,
+    input_descriptors: Vec<InputDescriptor>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    submission_requirements: Option<Vec<SubmissionRequirement>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    purpose: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    format: Option<ClaimFormatMap>,
+}
```

Comment:
As above, can use `Vec<ClaimFormat>`

Reply:
Removed the `Option` and but maintained `ClaimFormatMap` to pass test suite.